### PR TITLE
Add support for listing non-EC2 (Hybrid) instances

### PR DESCRIPTION
**Pull request by tedsmitt**
For https://github.com/gjbae1212/gossm/issues/5

Changed `findInstances` function so that it no longer discards instances that aren't EC2 (i.e. ResourceType: ManagedInstance)

The "Name" value of the non-EC2 instances is derived from the Tag:Name value of the managed instance in SSM.

Test output
```
root@ed9414f54764:/app/gossm/cmd# go test -v
=== RUN   TestExecute
gossm is interactive CLI tool that you select server in AWS and then could connect or send files your AWS server using start-session, ssh, scp in AWS Systems Manger Session Manager.

Usage:
  gossm [command]

Available Commands:
  cmd         Exec `run command` under AWS SSM with interactive CLI
  help        Help about any command
  mfa         It's to authenticate MFA on AWS, and save authenticated mfa token in .aws/credentials_mfa.
  scp         Exec `scp` under AWS SSM with interactive CLI
  ssh         Exec `ssh` under AWS SSM with interactive CLI
  start       Exec `start-session` under AWS SSM with interactive CLI

Flags:
  -c, --cred string      aws credentials file (default is $HOME/.aws/.credentials)
  -h, --help             help for gossm
  -p, --profile string   
                         [optional] if you are having multiple aws profiles, it is one of profiles (default is AWS_PROFILE environment variable or default)
  -r, --region string    [optional] it is region in AWS that would like to do something
  -t, --target string    [optional] it is instanceId of server in AWS that would like to something
  -v, --version          version for gossm

Use "gossm [command] --help" for more information about a command.
--- PASS: TestExecute (0.00s)
PASS
ok      github.com/gjbae1212/gossm/cmd  0.029s
```

### CODE REVIEW

File: cmd/internal.go [hunk 1]
```diff
@@ -335,17 +335,31 @@ func askMultiTarget(sess *session.Session, region string) (targets, domains []st
...
+	var ec2InstanceIds []string                    // used in the DescribeInstances call to filter results
+	var nonEc2Instances []*ssm.InstanceInformation // used to display any non-EC2 managed instances
 
-	// get managed instances
-	insts, err := findManagedInstances(sess, region)
-	if err != nil || len(insts) == 0 {
-	} else { // if instances exist.
-		input.Filters = append(input.Filters, &ec2.Filter{Name: aws.String("instance-id"), Values: aws.StringSlice(insts)})
+	managedInstances, err := findManagedInstances(sess, region) // get all ssm connected instances
+	if err != nil || len(managedInstances) == 0 {
```
Comment (gjbae1212) [inline cmd/internal.go:342]: @tedsmitt 
A one of the cases raised errors is to lack  "ssm:DescribeInstanceInformation" on AWS permissions.

In a specific situation, "ssm:DescribeInstanceInformation" permmision is redundant.

So if err ! =nil,  currently bypasses a filter such as aws.String("instance-id").

Your commit doesn't show instances list to users without "ssm:DescribeInstanceInformation".

**Review by gjbae1212 — COMMENTED** [review]
@tedsmitt 
Thank you, your contributes. 

I would say it needs to refine it. (about below comment).

File: cmd/internal.go (hunk 1 above)
Comment (tedsmitt) [reply cmd/internal.go:342]: Thanks @gjbae1212, I'll amend and test this now

**Comment by tedsmitt** [comment]
@gjbae1212 Thank you for your feedback! 😁

I've amended so that users will still get a list of instances without the `ssm:DescribeInstanceInformation` permission. I've tested this with a user by adding and removing the permission and checking the results. Looks to work ok now!

**Review by gjbae1212 — APPROVED** [review]
Thank you. 
LGTM.

**Comment by Bakrog** [comment]
Hi @gjbae1212 , I think this functionality got broken on master. I can't connect to my hybrid instances through gossm.

**Pull request merged**